Compilation error in http_integration.cc

The `HttpIntegrationTest::waitForNextUpstreamRequest` in test/integration/http_integration.cc has the following code:

```
  uint64_t upstream_with_request;
  if (!fake_upstream_connection_) {
    ...
        upstream_with_request = upstream_index;
    ...
  }
  ...
  return upstream_with_request;
```

and it fails to compile with
```
test/integration/http_integration.cc:364:10: error: 'upstream_with_request' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   return upstream_with_request;
          ^~~~~~~~~~~~~~~~~~~~~
```

(gcc, opt build)

--- a/test/integration/http_integration.cc
+++ b/test/integration/http_integration.cc
@@ -334,10 +334,10 @@ void HttpIntegrationTest::verifyResponse(IntegrationStreamDecoderPtr response,
   EXPECT_EQ(response->body(), expected_body);
 }
 
-uint64_t
+absl::optional<uint64_t>
 HttpIntegrationTest::waitForNextUpstreamRequest(const std::vector<uint64_t>& upstream_indices,
                                                 std::chrono::milliseconds connection_wait_timeout) {
-  uint64_t upstream_with_request = fake_upstreams_.size();
+  absl::optional<uint64_t> upstream_with_request;
   // If there is no upstream connection, wait for it to be established.
   if (!fake_upstream_connection_) {
 

--- a/test/integration/http_integration.cc
+++ b/test/integration/http_integration.cc
@@ -337,7 +337,7 @@ void HttpIntegrationTest::verifyResponse(IntegrationStreamDecoderPtr response,
 uint64_t
 HttpIntegrationTest::waitForNextUpstreamRequest(const std::vector<uint64_t>& upstream_indices,
                                                 std::chrono::milliseconds connection_wait_timeout) {
-  uint64_t upstream_with_request;
+  uint64_t upstream_with_request = fake_upstreams_.size();
   // If there is no upstream connection, wait for it to be established.
   if (!fake_upstream_connection_) {
 

--- a/test/integration/http_integration.h
+++ b/test/integration/http_integration.h
@@ -131,7 +131,7 @@ protected:
   // Sets fake_upstream_connection_ to the connection and upstream_request_ to stream.
   // In cases where the upstream that will receive the request is not deterministic, a second
   // upstream index may be provided, in which case both upstreams will be checked for requests.
-  uint64_t waitForNextUpstreamRequest(
+  absl::optional<uint64_t> waitForNextUpstreamRequest(
       const std::vector<uint64_t>& upstream_indices,
       std::chrono::milliseconds connection_wait_timeout = TestUtility::DefaultTimeout);
   void waitForNextUpstreamRequest(

--- a/test/integration/protocol_integration_test.cc
+++ b/test/integration/protocol_integration_test.cc
@@ -422,6 +422,7 @@ TEST_P(DownstreamProtocolIntegrationTest, RetryHostPredicateFilter) {
 
   // Note how we're expecting each upstream request to hit the same upstream.
   auto upstream_idx = waitForNextUpstreamRequest({0, 1});
+  ASSERT_LT(upstream_idx, fake_upstreams_.size());
   upstream_request_->encodeHeaders(Http::TestHeaderMapImpl{{":status", "503"}}, false);
 
   if (fake_upstreams_[upstream_idx]->httpType() == FakeHttpConnection::Type::HTTP1) {

--- a/test/integration/protocol_integration_test.cc
+++ b/test/integration/protocol_integration_test.cc
@@ -422,18 +422,18 @@ TEST_P(DownstreamProtocolIntegrationTest, RetryHostPredicateFilter) {
 
   // Note how we're expecting each upstream request to hit the same upstream.
   auto upstream_idx = waitForNextUpstreamRequest({0, 1});
-  ASSERT_LT(upstream_idx, fake_upstreams_.size());
+  ASSERT_TRUE(upstream_idx.has_value());
   upstream_request_->encodeHeaders(Http::TestHeaderMapImpl{{":status", "503"}}, false);
 
-  if (fake_upstreams_[upstream_idx]->httpType() == FakeHttpConnection::Type::HTTP1) {
+  if (fake_upstreams_[*upstream_idx]->httpType() == FakeHttpConnection::Type::HTTP1) {
     ASSERT_TRUE(fake_upstream_connection_->waitForDisconnect());
-    ASSERT_TRUE(fake_upstreams_[upstream_idx]->waitForHttpConnection(*dispatcher_,
-                                                                     fake_upstream_connection_));
+    ASSERT_TRUE(fake_upstreams_[*upstream_idx]->waitForHttpConnection(*dispatcher_,
+                                                                      fake_upstream_connection_));
   } else {
     ASSERT_TRUE(upstream_request_->waitForReset());
   }
 
-  waitForNextUpstreamRequest(upstream_idx);
+  waitForNextUpstreamRequest(*upstream_idx);
   upstream_request_->encodeHeaders(default_response_headers_, false);
   upstream_request_->encodeData(512, true);
 

